GOVERNANCE.md

```diff
@@ -0,0 +1,477 @@
+that specifically means:
+
+> While SemVer says there is no compatibility before 1.0.0, Cargo considers
+> 0.x.y to be compatible with 0.x.z, where y ≥ z and x > 0.
+
+Parts of the API that are covered by semver are explicitly marked. Our public
+API is extensively tested in integration tests. If changes to the public API of
+thin-egde.io are necessary, these changes are documented in the `CHANGELOG.md`
+file of the project, with reasoning why the change was necessary and how users
+can upgrade to the new interface(s).
+
+<!-- TODO: Expand on how exactly these changes are documented? -->
+
+We also define the following rules for our release process:
+
+* We release new features by increasing the "minor" version
+* When we release new patch releases, these releases never contain new features,
+  but only fixes to bugs.
+
+### Release cycle
+
+Minor releases are published roughly every TBD months.
+
+Until a minor release is published, the previous minor release _can_ receive
+patch releases. These patch releases _only_ contain bugfixes (accordingly to
+[semver](https://semver.org/)) and never contain new functionality.
+
+### Strategy
+
+We use the release-branch strategy, where development happens on the `main`
+branch and releases are done on dedicated `release-X.Y.Z` branches.
+
+Because we employ the "evergreen main branch" strategy and all of our merge
+commits on the `main` branch are guaranteed to successfully build and pass CI, a
+release can potentially be made starting from every commit on the `main` branch.
+
+For a description of the release workflow, read below.
+
+### Release maintainer
+
+For every release, one or more "release maintainers" are appointed. These
+maintainers have to be in the "release-maintainers" team.
+
+Multiple release maintainers should be prefered, to prevent release delay due to
+vacation, sickness or similar things.
+
+The responsibilities of the release maintainers are:
+
+* Creation and maintenance of the `release-x.y.z` branches
+* Cherry-picking bugfix patches from the `main` branch and applying them to the
+  relevant `release-x.y.z` branch (via pull request to that branch)
+  * If a patch does not apply cleanly, it is _not_ the responsibility of the
+    release maintainer to make the patch apply. It is responsibility of the
+    release maintainer, though, to talk to the developer of the patch and work
+    with them to port the fix back to the release
+* Create git-tags for the release(s) as appropriate
+* Create github artifacts (or take care of a github-action pipeline that does)
+  for releases
+* Port the changes to the `CHANGELOG.md` file back to `main` (via pull request)
+* Submit pull requests to the `main` branch to update all version numbers for
+  the next release cycle
+
+It is explicitely allowed for everyone to open bugfix-backport pull requests to
+a release branch, but the release maintainer(s) decide when and how to apply
+them to the release branch.
+
```

Comment:
```suggestion
```

Maybe we can just remove this bit, because it is not that important.

Reply:
I let it for now.